epochStart/bootstrap/process.go

```diff
@@ -181,27 +186,21 @@ func NewEpochStartBootstrap(args ArgsEpochStartBootstrap) (*epochStartBootstrap,
 		trieStorageManagers:        args.TrieStorageManagers,
 		uint64Converter:            args.Uint64Converter,
 		nodeShuffler:               args.NodeShuffler,
+		rounder:                    args.Rounder,
 	}
 
 	return epochStartProvider, nil
 }
 
-func (e *epochStartBootstrap) computedDurationOfEpoch() time.Duration {
-	return time.Duration(e.genesisNodesConfig.GetRoundDuration()*
-		uint64(e.generalConfig.EpochStartConfig.RoundsPerEpoch)) * time.Millisecond
-}
-
 func (e *epochStartBootstrap) isStartInEpochZero() bool {
 	startTime := time.Unix(e.genesisNodesConfig.GetStartTime(), 0)
 	isCurrentTimeBeforeGenesis := time.Since(startTime) < 0
 	if isCurrentTimeBeforeGenesis {
 		return true
 	}
 
-	configuredDurationOfEpoch := startTime.Add(e.computedDurationOfEpoch())
-	isEpochZero := time.Since(configuredDurationOfEpoch) < 0
-
-	return isEpochZero
+	currentRound := e.rounder.Index()
+	return float64(currentRound) < float64(e.generalConfig.EpochStartConfig.RoundsPerEpoch)*gracePeriodInPercentage
```

Comment:
this won't work with the early start of epoch. Can remain like this for time being.

Reply:
yep. but that is hard to know in this case.

Reply:
agree, probably we will have a dedicated component for this